### DISCUSSION TITLE
Manual Docker update in SO (v)2.4.100.

### DISCUSSION THREAD

**Discussion by virginiait** [Security-Onion-Solutions/securityonion#13903] [Unsupported Versions] [answered]
**Version**
2.4.100

**Installation Method**
ISO

**Description**
configuration

**Installation Type**
Standalone

**Location**
On-prem. Airgap.

**Hardware Specs**
Exceeds minimums

**CPU**
16

**RAM**
64

**Storage for /**
1 TB

**Storage for /nsm**
1 TB

**Network Traffic Collection**
Inline - TAP

**Network Traffic Speeds**
1Gbps

**Status**
OK

**Detail**

How should I implement the update to Docker 27.2, until we're able to implement the newer SO version?

Attempting to manually update 2.4.110 installed Docker v26.1.4 to v27.2.x. Reason - installing the update until we're able to migrate to the latest SO version. The manual update to Docker v27.2.x releaves findings identified in CVE-2024-411120 and acceptance gates from decision stakeholders. I did see the update added to the (v)2.4.110 baseline: https://github.com/Security-Onion-Solutions/securityonion/issues/13566. However, it will be some time before we're able to implement future SO versions. To update, the following files were installed via both methods below (seperatly of course):
  sudo rpm -Uvh docker-ce*
     later ran
 sudo dnf install docker-ce-*
docker-ce-cli-27.2.0-1.el9.x86_64
docker-ce-rootless-extras-27.2.0-1.el9.x86_64
docker-ce-27.2.0-1.el9.x86_64

Install verified with:
#rpm -qa | grep docker-ce

However, shortly there after and reboots - the previous version showed back up. Assuming salt brought it back.

How should I implement the update to Docker 27.2, until we're able to implement the newer SO version?

Many Thanks.

**Comment by cm-ops** [comment, marked as answer]
Docker and salt versions are version locked and only upgraded via soup after testing. You may be able to get this to work but upgrading componenets outside of soup is not recommended.

**Reply by virginiait** [reply]
Thanks. Appreciate all of your team's efforts.
